pkg/updater/updater.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package updater implements the updater.
+package updater
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/updater/repository"
+)
+
+const (
+	// defaultRepositoryPath is the default path to the repository.
+	defaultRepositoryPath = "/opt/datadog-packages"
+)
+
+// Install installs the default version for the given package.
+// It is purposefully not part of the updater to avoid misuse.
+func Install(ctx context.Context, orgConfig *OrgConfig, pkg string) error {
+	downloader := newDownloader(http.DefaultClient)
+	repository := &repository.Repository{RootPath: path.Join(defaultRepositoryPath, pkg)}
+	firstPackage, err := orgConfig.GetDefaultPackage(ctx, pkg)
+	if err != nil {
+		return fmt.Errorf("could not get default package: %w", err)
+	}
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return fmt.Errorf("could not create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	err = downloader.Download(ctx, firstPackage, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not download package: %w", err)
+	}
+	err = repository.Create(firstPackage.Version, tmpDir)
```

Comment:
Do we use any hashes/signatures on these to ensure integrity?

Reply:
Can you be a bit more explicit on which part/data you are asking this ?

As a summary the way we are doing integrity here is:
- A `catalog.json` is sent over RC using RC's existing TUF-based integrity framework. Without getting into the technical details we have a guarentee that this file was signed by a key dedicated to this `catalog.json` file.
- Each package version in that catalog has an associated `sha256` hash
- This hash is checked against the downloaded archive before we take any action on it

Note that currently `catalog.json` is hardcoded and that the RC bits are not implemented yet.

I have docs to share on slack if you want more details about either RC itself or this mechanism.

Reply:
If we do TUF for the catalog, then my main concern is around gaps between verification and use of repo. Either way, we can talk on slack if needed and I'd love to read up on the implementation details!

Reply:
Sure, there's a link for the design in fleet-automation-update on slack. You can get a link to the security-focused doc from there. Feel free to ping if you have issues finding it or questions